src/Items/ItemCactus.h

```diff
@@ -0,0 +1,68 @@
+
+#pragma once
+
+#include "ItemHandler.h"
+
+
+
+class cItemCactusHandler :
+	public cItemHandler
+{
+public:
+	cItemCactusHandler(int a_ItemType) :
+		cItemHandler(a_ItemType)
+	{
+	}
+
+
+	virtual bool IsPlaceable(void) override
+	{
+		return true;
+	}
+
+
+	virtual bool OnPlayerPlace(
+        cWorld & a_World, cPlayer & a_Player, const cItem & a_EquippedItem,
+        int a_BlockX, int a_BlockY, int a_BlockZ, eBlockFace a_BlockFace,
+        int a_CursorX, int a_CursorY, int a_CursorZ
+    ) 
+	{
+		/** We will use the block face and the block coordinates to determine if we can place the cactus
+		There must be sand or cactus under the cactus and no blocks on the sides inorder for it to be placed
+		*/
+		Vector3i PlacementPos = GetBlockNextTo(Vector3i(a_BlockX, a_BlockY, a_BlockZ), a_BlockFace);
```

Comment:
Prefer to use `auto` as much as possible.

Reply:
Is this true? I haven't seen auto used anywhere else

Reply:
It's an old codebase, so it's not used everywhere, but we try to use it in all new code.